src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -1597,10 +1597,8 @@ def calc_slope(data_sect, gdq_sect, frame_time, opt_res, save_opt, rn_sect,
     err_2d_array = data_sect[0, :, :] * frame_time
 
     # Suppress, then re-enable, harmless arithmetic warnings
-    '''
```

Comment:
You could use Numpy's `errstate` context manager here:

```python
with np.errstate(divide="ignore", invalid="ignore"):
    err_2d_array[err_2d_array < 0] = 0
```

The current error/warning global config for numpy can be grabbed via `np.geterr()`.

Reply:
The question is why is e.g. `divide by zero`  filtered out. Is there a chance it's going to be raised?
But since this is not related to the current bug being fixed I suggest if necessary it can be changed in another PR.

Reply:
I removed the code changes not directly related to the bug in JP-1950.